triton/tut/parallel.rst

```diff
@@ -214,122 +213,91 @@ you should load the same compiler that you used for compiling the code.
-
-  - OpenMPI (e.g. ``openmpi/3.1.4``)
-  - Intel's MPI (e.g. ``intel-parallel-studio/cluster.2020.0-intelmpi``)
+MPI standard. We recommend that you use our OpenMPI installation
+(``openmpi/4.0.5``). For information on other installed versions, see the
+:doc:`MPI applications-page<../apps/mpi>`.
 
-Some libraries/programs might have already existing requirement for a certain
-MPI version. If so, use that version or ask for administrators to create
-a version of the library with dependency on the MPI version you require.
-
-.. warning::
-
-   Different versions of MPI are not compatible with each other. Each
-   version of MPI will create code that will run correctly with only
-   that version of MPI. Thus if you create code with a certain version,
-   you will need to load the same version of the library when you are
-   running the code.
-
-   Also, the MPI libraries are usually linked to slurm and network
-   drivers. Thus, when slurm or driver versions are updated, some
-   older versions of MPI might break. If you're still using said
-   versions, let us know. If you're just starting a new project, it
-   is recommended to use our recommended MPI libraries.
+.. include:: /triton/ref/mpi-warning.rst
 
 For basic use of MPI programs, you will need to use the
 ``-n N``/``--ntasks=N``-option to specify the number of MPI workers.
 
-Running a typical MPI program
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-The following use ``hpc-examples`` from
-:ref:`the previous exercises <triton-tut-exercise-repo>`.  Get the
-code::
-
-  git clone https://github.com/AaltoSciComp/hpc-examples/
-  cd hpc-examples/hello_mpi/          # C
-  cd hpc-examples/hello_mpi_fortran/  # fortran
+Compiling and running an MPI Hello world-program
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Loading module::
+.. include:: /triton/examples/mpi/hello.rst
 
-  # GCC + OpenMPI
-  module load gcc/9.2.0      # GCC
-  module load openmpi/3.1.4  # OpenMPI
-
-  # Intel compilers + Intel's MPI
-  module load intel-parallel-studio/cluster.2019.3-intelmpi
-
-Compiling the code (depending on module and language)::
-
-  # OpenMPI
-  mpicc    -O2 -g hello_mpi.c -o hello_mpi # C code
-  mpifort  -O2 -g hello_mpi_fortran.f90 -o hello_mpi_fortran # Fortran code
-
-  # Intel MPI
-  mpiicc   -O2 -g hello_mpi.c -o hello_mpi # C code
-  mpiifort -O2 -g hello_mpi_fortran.f90 -o hello_mpi_fortran # Fortran code
-
-Running the program with srun (for testing)::
+Spreading MPI workers evenly
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-  srun --time=00:05:00 --mem-per-cpu=200M --ntasks=4 ./hello_mpi
+In many cases you might require more than one node during your job's runtime.
 
-Running an MPI code in the batch mode:
+When this is the case, it is usually recommended to split the number of
+workers evenly among the nodes. To do this, one can use
+``-N N``/``--nodes=N`` and ``--ntasks-per-node=n``. For example, you could
+distribute the previously requested four tasks to two nodes with:
```

Comment:
What's the actual advantage of this? i.e. balancing on different nodes. Your CPU power will stay the same (or be different depending on the node), and the only direct advantage I can see is not to be killed as quick if you run out of your memory quota...